torch/testing/_internal/common_device_type.py

```diff
@@ -574,6 +574,22 @@ def __call__(self, fn):
         return fn
 
 
+tfloat32 = "_testing.tfloat32"
+tcomplex64 = "_testing.tcomplex64"
+
+
+def tf32_to_fp32(dtype):
+    if dtype == tfloat32:
+        return torch.float32
+    elif dtype == tcomplex64:
```

Comment:
Really? This works for complex?

Reply:
Yeah, the TF32 math mode applies to fp32 and complex64, from what I see in failed tests.